machine/v1beta1/types_azureprovider.go

```diff
@@ -126,6 +132,56 @@ type SpotVMOptions struct {
 	MaxPrice *resource.Quantity `json:"maxPrice,omitempty"`
 }
 
+// AzureDiagnostics is used to configure the diagnostic settings of the virtual machine.
+type AzureDiagnostics struct {
+	// AzureBootDiagnostics configures the boot diagnostics settings for the virtual machine.
+	// This allows you to configure capturing serial output from the virtual machine on boot.
+	// This is useful for debugging software based launch issues.
+	// + This is a pointer so that we can validate required fields only when the structure is
```

Comment:
This is something I wanted to discuss with a wider audience.

The installer sets this up for masters using a storage account that it creates for the cluster (the name of which we don't have access to within the cluster as far as I can tell).

We could turn this on by default in an Azure managed storage account, it would cost $0.05/VM/Month for each cluster.
I was leaning towards not enabling it by default as it adds cost, but I appreciate it's negligible in the scheme of things.

Also, if we do enable by default, we may want to find a way to pass in the configuration from the installer (or we could add this feature and have the installer set it by default going forward, allowing users to opt out)